### PR TITLE
Adds test edge cases for text generation

### PR DESCRIPTION
**Pull request by flaviabeo**
This PR is regarding adding tests for edge cases where the text sent for generation is a white space or has special characters. This compliments the work for the detectors test cases at #97. 

Testing straight to the service, both cases included work:
1. Generation text with whitespace as input:
```json
{
    "inputs": "   ",
    "model_id": "bloom-560m",
    "text_gen_parameters": {
        "preserve_input_text": true,
        "max_new_tokens": 99,
        "min_new_tokens": 1,
        "truncate_input_tokens": 500,
        "decoding_method": "SAMPLING",
        "top_k": 2,
        "top_p": 0.9,
        "typical_p": 0.5,
        "temperature": 0.8,
        "seed": 42,
        "repetition_penalty": 2,
        "max_time": 0,
        "stop_sequences": [
            "42"
        ]
    }
}

### response is:

{
	"generated_text": "    = \"(\" + (i * 100) / 1000.0  +\", \"+ i % 10 );\n    }\n\n    public static void main() {\n        System .out   += new String[0] ;\n\n        for(  int i=0; i<10; i++ )\n            if( (System    ).compareToIgnoreCase((String)  ) )\n                break;\n            else \n                i++;\n\n            //println  (i+1));//printing the value of this variable in a string\n\n}\n",
	"token_classification_results": {},
	"finish_reason": "EOS_TOKEN",
	"generated_token_count": 90,
	"seed": 0,
	"input_token_count": 1
}
```
2. Generation text with special characters as input:

```json
{
    "inputs": "!!!",
    "model_id": "bloom-560m",
    "text_gen_parameters": {
        "preserve_input_text": true,
        "max_new_tokens": 99,
        "min_new_tokens": 1,
        "truncate_input_tokens": 500,
        "decoding_method": "SAMPLING",
        "top_k": 2,
        "top_p": 0.9,
        "typical_p": 0.5,
        "temperature": 0.8,
        "seed": 42,
        "repetition_penalty": 2,
        "max_time": 0,
        "stop_sequences": [
            "42"
        ]
    }
}
### response is:
{
	"generated_text": "!!!\nThe first thing I did was to get a new one. The old ones were not so good, and the newer are better.\nI have been using this for about 2 years now (and it is still working great). It has worked well in my house since then!\nIt does work on some of our windows but most do NOT! So if you want your window tinted or replaced with something that works just fine without having any problems at all please call us today PI:KEY",
	"token_classification_results": {},
	"finish_reason": "EOS_TOKEN",
	"generated_token_count": 98,
	"seed": 0,
	"input_token_count": 1
}
```

### CODE REVIEW

File: src/clients/nlp.rs [hunk 1]
```diff
@@ -114,3 +116,178 @@ fn request_with_model_id<T>(request: T, model_id: &str) -> Request<T> {
...
+                finish_reason: 2,
+                input_token_count: 20,
+                seed: 23,
+                tokens: vec![gen_token],
+                input_tokens: vec![input_token],
+            }));
+        assert!(mock_nlp_client
+            .text_generation_task_predict("bloom-560m", request)
+            .await
+            .is_ok());
```
[the remark this answers is not in the document]
Comment (flaviabeo) [reply src/clients/nlp.rs:291]: I guess I will leave only one case in this file, and then add the tests for the edge cases at the orchestrator. I was trying to understand what was the best place to do them.

File: src/orchestrator.rs [hunk 2]
```diff
@@ -982,6 +982,118 @@ mod tests {
...
+    #[tokio::test]
+    async fn test_tgis_generate_whitespace() {
+        // Initialize a mock object from `TgisClient`
+        let mut mock_client = TgisClient::faux();
+
+        let sample_text = String::from("   ");
+        let text_gen_model_id = String::from("test-llm-id-1");
+
+        let generation_response = GenerationResponse {
+            text: String::from("    sample response worked"),
```
[the remark this answers is not in the document]
Comment (flaviabeo) [reply src/orchestrator.rs:1002]: Will look into these. Thank you!!

**Comment by flaviabeo** [comment]
Closing this one to define better way to test this case.
